llvm/test/Transforms/InstCombine/ptrauth-intrinsics-call.ll

```diff
@@ -3,7 +3,9 @@
 
 define i32 @test_ptrauth_call_sign(ptr %p) {
 ; CHECK-LABEL: @test_ptrauth_call_sign(
-; CHECK-NEXT:    [[V3:%.*]] = call i32 [[P:%.*]]()
+; CHECK-NEXT:    [[V0:%.*]] = ptrtoint ptr [[P1:%.*]] to i64
+; CHECK-NEXT:    [[P:%.*]] = inttoptr i64 [[V0]] to ptr
+; CHECK-NEXT:    [[V3:%.*]] = call i32 [[P]]()
```

Comment:
This looks like a design defect in the ptrauth intrinsics -- why are they working on integers, not pointers?